hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/table/catalog/HiveSchemaUtils.java

```diff
@@ -177,10 +180,19 @@ private static DataType toFlinkPrimitiveType(PrimitiveTypeInfo hiveType) {
 
   /**
    * Create Hive field schemas from Flink table schema including the hoodie metadata fields.
+   *
+   * @param table
    */
-  public static List<FieldSchema> toHiveFieldSchema(TableSchema schema) {
+  public static List<FieldSchema> toHiveFieldSchema(CatalogBaseTable table) {
+    TableSchema schema = table.getSchema();
+    Configuration configuration = Configuration.fromMap(table.getOptions());
+    Boolean changelogEnable = configuration.getBoolean(FlinkOptions.CHANGELOG_ENABLED);
+    Collection<String> hoodieMetaColumns = HoodieRecord.HOODIE_META_COLUMNS;
+    if (changelogEnable) {
+      hoodieMetaColumns = HoodieRecord.HOODIE_META_COLUMNS_WITH_OPERATION;
```

Comment:
In current master, we do not add _hoodie_operation field for hive table now, how the hive table was created locally ?

Reply:
specifying changlog.enabled=true and hive_sync.skip_ro_suffix = true， and Hive sync automatically synchronizes metadata

Reply:
The `_hoodie_operation` field is useless for hive table, we should not sync it.

Reply:
What version of hudi did you use for the streaming writer ?

Reply:
> What version of hudi did you use for the streaming writer ?

0.12.1

Reply:
Can you past the error stack trace here ? The writer does not expect to sync the `_hoodie_operation` meta field now.

Reply:
![image](https://user-images.githubusercontent.com/59957056/198202085-65569dd0-1abe-44a3-bc8a-f330d0174f88.png)

Reply:
This should be because the operation field exists in the parquet file and is not filtered out.